apollo-router/src/plugins/connectors/make_requests/graphql_utils.rs

```diff
@@ -0,0 +1,132 @@
+use apollo_compiler::executable::Field;
+use apollo_compiler::executable::Operation;
+use apollo_compiler::executable::Selection;
+use apollo_compiler::schema::Value;
+use apollo_compiler::Node;
+use serde_json::Number;
+use serde_json_bytes::ByteString;
+use serde_json_bytes::Map;
+use serde_json_bytes::Value as JSONValue;
+use tower::BoxError;
+
+use super::MakeRequestError;
+use super::ENTITIES;
+
+pub(super) fn field_arguments_map(
+    field: &Node<Field>,
+    variables: &Map<ByteString, JSONValue>,
+) -> Result<Map<ByteString, JSONValue>, BoxError> {
+    let mut arguments = Map::new();
+
+    for argument in field.arguments.iter() {
+        match &*argument.value {
+            apollo_compiler::schema::Value::Variable(name) => {
+                if let Some(value) = variables.get(name.as_str()) {
+                    arguments.insert(argument.name.as_str(), value.clone());
+                }
+            }
+            _ => {
+                arguments.insert(
+                    argument.name.as_str(),
+                    argument_value_to_json(&argument.value)?,
+                );
+            }
+        }
+    }
+
+    for argument_def in field.definition.arguments.iter() {
+        if let Some(value) = argument_def.default_value.as_ref() {
+            arguments
+                .entry(argument_def.name.as_str())
+                .or_insert_with(|| {
+                    argument_value_to_json(value).unwrap_or_else(|err| {
+                        tracing::warn!("failed to convert default value to json: {}", err);
```

Comment:
Nitpick: this would result in a very vague warning like `failed to convert default value to json: invalid int`. It'd be helpful to have the name of the field and argument.

Reply:
added!